test/extended/cli/resources.go

```diff
@@ -176,3 +179,46 @@ func writeObjectToFile(obj runtime.Object) (string, error) {
 
 	return name, nil
 }
+
+// replaceImageInFile reads in a file, replaces the given image name, then writes to a new temporary file.
+func replaceImageInFile(source, oldImage, newImage string) (string, error) {
```

Comment:
I'm not a fan of manually replacing like so, I'd suggest using templates similarly how we do that upstream, see: https://github.com/openshift/origin/blob/e000efe2de764b8f0789bbea7bc2ebe248eb5275/vendor/k8s.io/kubernetes/test/e2e/kubectl/kubectl.go#L296-L303